test/components/Documents/DocumentTable.test.jsx

```diff
@@ -1,63 +1,107 @@
-        endDate: null,
-        uploadDate: new Date(),
-        fileUrl: 'http://localhost:3000/pod/PASS/Documents/test2.pdf'
-      }
-    ],
-    dataset: {},
-    containerURL: 'foo'
-  },
-  loadingDocuments: false
-};
+// Clear created DOM after each test
+afterEach(() => {
+  cleanup();
+});
+
+const MockTableComponent = ({
+  documentListObject,
+  loadingDocuments,
+  isMobile,
+  isLoggedIn = true
+}) => {
+  window.matchMedia = createMatchMedia(isMobile ? 500 : 1200);
 
-const MockTableComponent = () => (
-  <BrowserRouter>
-    <DocumentListContext.Provider value={mockedDocumentContext}>
-      <DocumentTable />
+  return render(
+    <DocumentListContext.Provider value={{ documentListObject, loadingDocuments }}>
+      <BrowserRouter>
+        <DocumentTable
+          handleAclPermissionsModal={vi.fn()}
+          handleSelectDeleteDoc={vi.fn()}
+          session={{ info: { isLoggedIn } }}
+        />
+      </BrowserRouter>
     </DocumentListContext.Provider>
-  </BrowserRouter>
-);
+  );
+};
+
+const documentListObject = {
+  docList: [
+    {
+      id: '1',
+      name: 'test.pdf',
+      type: 'Other',
+      description: 'test description',
+      uploadDate: new Date(),
+      endDate: new Date(),
+      fileUrl: 'http://localhost:3000/pod/PASS/Documents/test.pdf'
+    },
+    {
+      id: '2',
+      name: 'test2.pdf',
+      type: 'Passport',
+      description: 'test description 2',
+      uploadDate: new Date(),
+      endDate: null,
+      fileUrl: 'http://localhost:3000/pod/PASS/Documents/test2.pdf'
+    }
+  ]
+};
 
 describe('DocumentTable Component', () => {
-  it('renders with documents', () => {
-    const { getAllByRole, queryByRole } = render(<MockTableComponent />);
-    const allRows = getAllByRole('row');
-    expect(allRows.length).toBe(3);
-
-    const row1FileName = queryByRole('cell', { name: 'test.pdf' });
-    const row1FileType = queryByRole('cell', { name: 'Other' });
-    const row1Description = queryByRole('cell', { name: 'test description' });
-    const row2FileName = queryByRole('cell', { name: 'test2.pdf' });
-    const row2FileType = queryByRole('cell', { name: 'Passport' });
-    const row2Description = queryByRole('cell', { name: 'test description 2' });
-
-    expect(row1FileName).not.toBeNull();
-    expect(row1FileType).not.toBeNull();
-    expect(row1Description).not.toBeNull();
-    expect(row2FileName).not.toBeNull();
-    expect(row2FileType).not.toBeNull();
-    expect(row2Description).not.toBeNull();
+  it('renders without crashing', () => {
+    MockTableComponent({ documentListObject, loadingDocuments: false });
+    expect(screen.getByText('test.pdf')).toBeInTheDocument();
+  });
+
+  it('displays loading animation when documents are loading', () => {
+    MockTableComponent({ documentListObject, loadingDocuments: true });
+    expect(screen.getByTestId('loading-animation')).toBeInTheDocument();
+  });
+
+  it('renders DocumentsDesktop for larger screens', () => {
```

Comment:
Aren't these two tests very similar to the final two?

Reply:
I was wondering where our conventional balance between redundancy and thoroughness lies. I may have opted for the latter in more cases than necessary. I'll look into filtering moot redundancy out.

Reply:
Yes I see it now, they were actually nearly identical. Just something I'd overlooked.